README.md

```diff
@@ -254,8 +254,135 @@ class MysqlSpec extends FlatSpec with ForAllTestContainer {
 }
 ```
 
+## New API
+
+Starting from 0.34.0 version testcontainers-scala provides the new API. 
+The main motivation points are in the [pull request](https://github.com/testcontainers/testcontainers-scala/pull/78).
+
+### `Container` and `ContainerDef` 
+
+Docker containers are represented through the two different entities:
+1. `ContainerDef` — it's container definition. `ContainerDef` describes, how to build a container.
+   You can think about it like about a container constructor, or dockerfile description.
+   Usually, `ContainerDef` receives some parameters.
+   `ContainerDef` has a `start()` method. It returns a started `Container`.
+2. `Container` — it's a started container. You can interact with it through its methods.
+   For example, in the case of `MySQLContainer` you can get it's JDBC URL with `jdbcUrl` method.
+   `Container` is the main entity for using inside tests.
+
+### Scalatest usage
+
+You can use one of the four traits:
+1. `TestContainerForAll` — will start a single container before all tests and stop after all tests.
+2. `TestContainerForEach` — will start a single container before each test and stop after each test.
+3. `TestContainersForAll` — will start multiple containers before all tests and stop after all tests.
+4. `TestContainersForEach` — will start multiple containers before each test and stop after each test.
+
+
+#### Single container in tests
+
+If you want to use a single container in your test:
+```scala
+class MysqlSpec extends FlatSpec with TestContainerForAll {
+
+  // You need to override `containerDef` with needed container definition
+  override val containerDef = MySQLContainer.Def()
```

Comment:
I still believe that main container class should be its definition.
```
override val container = MySQLContainer()
```
to make it more intuitive

Reply:
I played with this a bit and I didn't come up with something reasonable enough. Going this way will make it much harder to maintain backward compatibility with the current API. In this pull request, I managed to maintain backward compatibility relatively easy.

Anyway, this is experimental. I suggest to try it and wait for feedback.

Reply:
Makes sense. I think I need to play with the code on my local machine to get a better sense of the backward compatibility in this implementation.

Btw, do you know any good way to gather feedback from early adopters?

Reply:
I definitely will promote these changes in the [Russian speaking scala user group in the telegram](https://t.me/scala_ru). I know that some people there are using testcontainers.

Also, recently I spoke about testcontainers and this pull request in the [scalalaz podcast](https://scalalaz.ru/). This episode is still not released though. We need to wait for it a few more days.

I hope it would be enough at the start.